src/core/mount.c

```diff
@@ -274,13 +275,35 @@ static int update_parameters_proc_self_mountinfo(
 
         p = &m->parameters_proc_self_mountinfo;
 
-        r = free_and_strdup(&p->what, what);
-        if (r < 0)
-                return r;
+        if (what && !utf8_is_valid(what)) {
+                _cleanup_free_ char *what_escaped = NULL;
 
-        q = free_and_strdup(&p->options, options);
-        if (q < 0)
-                return q;
+                what_escaped = utf8_escape_invalid(what);
+                if (!what_escaped)
+                        return -ENOMEM;
+
+                free_and_replace(p->what, what_escaped);
+                r = 1;
```

Comment:
No no, please always compare with the previous value.

Reply:
What do you mean "compare with previous value"?

Reply:
`p->what` may already contain the same (escaped) value. In that case, we should not return positive value.
```c
                what_escaped = utf8_escape_invalid(what);
                if (!what_escaped)
                        return -ENOMEM;

                r = streq_ptr(p->what, what_escaped);
                free_and_replace(p->what, what_escaped);
```

Reply:
Hmmm, you are right. We should return 0 when they are the same already even though we will be "replacing" the string.

Reply:
However,
```C
r  = streq_ptr(p->what, what_escaped);
```

seems incorrect. Instead it should be, 

```C
r = strneq(p->what, what_escaped);
```

Reply:
Ah, yes. I forgot `!`.